server/data_common/matrix_loader.py

```diff
@@ -19,32 +19,33 @@ def __init__(self, loader):
         self.data_adaptor = None
         self.data_lock = RWLock()
 
-    def acquire(self, app_config):
-        """returns the data_adaptor if cached.  opens the data_adaptor if not.
-        In either case, the a reader lock is taken.  Must call release when
-        the data_adaptor is no longer needed"""
-
+    def acquire_existing(self):
+        """If the data_adaptor exists, take a read lock and return it, else return None"""
         self.data_lock.r_acquire()
         if self.data_adaptor:
             return self.data_adaptor
 
         self.data_lock.r_release()
-        try:
-            with self.data_lock.w_locked():
-                # the data may have been loaded while waiting on the lock
-                if not self.data_adaptor:
-                    self.loader.pre_load_validation()
-                    self.data_adaptor = self.loader.open(app_config)
-
-        except Exception:
-            # necessary to acquire after an exception, since the release will occur when
-            # the context exits
-            self.data_lock.r_acquire()
-            raise
+        return None
 
+    def acquire(self, app_config):
```

Comment:
you might consider naming this something more descriptive, like `acquire_and_load()`

Reply:
Done, called it acquire_and_open